test/tests/materials/linear_spring/spring_static.i

```diff
@@ -0,0 +1,424 @@
+  [../]
+  [./rot_accel_z]
+    type = NewmarkAccelAux
+    variable = rot_accel_z
+    displacement = rot_z
+    velocity = rot_vel_z
+    beta = 0.25
+    execute_on = timestep_end
+  [../]
+  [./rot_vel_z]
+    type = NewmarkVelAux
+    variable = rot_vel_z
+    acceleration = rot_accel_z
+    gamma = 0.5
+    execute_on = timestep_end
+  [../]
+[]
+
+[BCs]
+  [./fixx1]
+    type = DirichletBC
+    variable = disp_x
+    boundary = left
+    value = 0.0
+  [../]
+  [./fixy1]
+    type = DirichletBC
+    variable = disp_y
+    boundary = left
+    value = 0.0
+  [../]
+  [./fixz1]
+    type = DirichletBC
+    variable = disp_z
+    boundary = left
+    value = 0.0
+  [../]
+  [./fixr1]
+    type = DirichletBC
+    variable = rot_x
+    boundary = left
+    value = 0.0
+  [../]
+  [./fixr2]
+    type = DirichletBC
+    variable = rot_y
+    boundary = left
+    value = 0.0
+  [../]
+  [./fixr3]
+    type = DirichletBC
+    variable = rot_z
+    boundary = left
+    value = 0.0
+  [../]
+[]
+
+[NodalKernels]
+  [./force_x]
+    type = UserForcingFunctionNodalKernel
+    variable = disp_x
+    boundary = right
+    function = force_x
+  [../]
+  [./force_y]
+    type = UserForcingFunctionNodalKernel
+    variable = disp_y
+    boundary = right
+    function = force_y
+  [../]
+  [./force_z]
+    type = UserForcingFunctionNodalKernel
+    variable = disp_z
+    boundary = right
+    function = force_z
+  [../]
+  [./moment_x]
+    type = UserForcingFunctionNodalKernel
+    variable = rot_x
+    boundary = right
+    function = moment_x
+  [../]
+  [./moment_y]
+    type = UserForcingFunctionNodalKernel
+    variable = rot_y
+    boundary = right
+    function = moment_y
+  [../]
+  [./moment_z]
+    type = UserForcingFunctionNodalKernel
+    variable = rot_z
+    boundary = right
+    function = moment_z
+  [../]
+[]
+
+[Functions]
+  [./force_x]
+    type = PiecewiseLinear
+    x = '0.0 1.0 2.0 3.0 4.0' # time
+    y = '0.0 1.0 0.0 -1.0 0.0'  # force
+  [../]
+  [./force_y]
+    type = PiecewiseLinear
+    x = '0.0 1.0 2.0 3.0 4.0' # time
+    y = '0.0 2.0 0.0 -2.0 0.0'  # force
+  [../]
+  [./force_z]
+    type = PiecewiseLinear
+    x = '0.0 1.0 2.0 3.0 4.0' # time
+    y = '0.0 3.0 0.0 -3.0 0.0'  # force
+  [../]
+  [./moment_x]
+    type = PiecewiseLinear
+    x = '0.0 1.0 2.0 3.0 4.0' # time
+    y = '0.0 10.0 0.0 -10.0 0.0'  # moment
+  [../]
+  [./moment_y]
+    type = PiecewiseLinear
+    x = '0.0 1.0 2.0 3.0 4.0' # time
+    y = '0.0 20.0 0.0 -20.0 0.0'  # moment
+  [../]
+  [./moment_z]
+    type = PiecewiseLinear
+    x = '0.0 1.0 2.0 3.0 4.0' # time
+    y = '0.0 30.0 0.0 -30.0 0.0'  # force
+  [../]
+[]
+
+[Materials]
+  [./linear_spring_test]
+    type = LinearSpring
+    block = 0
+    y_orientation = '0.0 1.0 0.0'
+    displacements = 'disp_x disp_y disp_z'
+    rotations = 'rot_x rot_y rot_z'
+    kx = 1.0
+    ky = 2.0
+    kz = 3.0
+    krx = 10.0
+    kry = 20.0
+    krz = 30.0
+  [../]
+  [./density]
```

Comment:
The density block is not used, right? Also, if this is a quasi-static test, then the vel and accel aux variables and the corresponding aux kernels are not required.